packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -916,6 +918,23 @@ function build(
     if (enableDisavow) {
       vpow.disavow = disavow;
     }
+    if (enableVatstore) {
+      vpow.vatstore = harden({
```

Comment:
let's add a comment here to describe how the vat's keyspace is carved up, and what code is responsible:

* userspace `vatstore`: `vvs.${userKey}`
* VOM `makeWeakStore`: `ws${storeID}.${vobjID}` (virtualObjectManager.js / makeWeakStore / virtualObjectKey)
* VOM virtual objects: `o+${kindID}/${instanceID}` (virtualObjectManager.js / makeKind / makeNewInstance)

Or, put this comment up at the top of liveslots.js, and add pointers to it from the three places that currently compete for keyspace.

My biggest concern is that we might accidentally add a new feature in the future, and the implementor doesn't realize the importance of keeping the keyspaces distinct, and we wind up with a breach enabled by userspace intentionally colliding with the keyspace of something more powerful.

Reply:
I'd feel slightly better if the virtual object manager used `vom.` in front of everything it manages (so `vom.ws${storeID}.${vobjID}` and `vom.o+${kindID}/${instanceID}`). On one hand, that feels overkill because those three keyspaces are already obviously non-overlapping. On the other hand, then a third feature can just choose a different prefix (`foo.`) instead of also needing to worry about how the VOM might change in the future and consume more of the keyspace.